### PR TITLE
schedule: fix address space mismatches

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Fix below sparse warnings:

zephyr_dp_schedule.c:134:20: warning: incorrect type in argument 1 (different address spaces)

zephyr_dp_schedule.c:348:36: warning: incorrect type in assignment (different address spaces)

introduced by 3ee1d787

### CODE REVIEW

**Comment by aiChaoSONG** [comment]
SOFCI TEST

**Review by dbaluta — APPROVED** [review]

File: src/schedule/zephyr_dp_schedule.c [hunk 1]
```diff
@@ -131,7 +131,7 @@ static int scheduler_dp_task_free(void *data, struct task *task)
 	scheduler_dp_unlock(lock_key);
 
 	/* free task stack */
-	rfree(pdata->p_stack);
+	rfree(z_soc_uncached_ptr(pdata->p_stack));
```
Comment (marcinszkudlinski) [inline src/schedule/zephyr_dp_schedule.c:134]: not needed, heap implementation works on both cached and uncached aliases. Changing to uncache here is just a waste of CPU cycles

File: src/schedule/zephyr_dp_schedule.c [hunk 2]
```diff
@@ -345,7 +345,7 @@ int scheduler_dp_task_init(struct task **task,
 	/* success, fill the structures */
 	task_memory->task.priv_data = &task_memory->pdata;
 	task_memory->pdata.thread_id = thread_id;
-	task_memory->pdata.p_stack = p_stack;
+	task_memory->pdata.p_stack = z_soc_cached_ptr(p_stack);
```
Comment (marcinszkudlinski) [inline src/schedule/zephyr_dp_schedule.c:348]: here too, alloc will return cached pointer. no need to change
to be 100% accurate - a flag SOF_MEM_CAPS_CACHE indeed may be added to allocation
It won't change a bit in flow, just be more clear to read

task_memory = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM **| SOF_MEM_CAPS_CACHE**, sizeof(*task_memory));


p_stack = rballoc_align(0, SOF_MEM_CAPS_RAM **| SOF_MEM_CAPS_CACHE**, stack_size, Z_KERNEL_STACK_OBJ_ALIGN);

**Review by marcinszkudlinski — CHANGES_REQUESTED** [review]
Good change to Zephyr kernel types no need however to play with cached/uncached aliases. Wasn't __sparse_cache macros dedicated to avoid warnings?

File: src/schedule/zephyr_dp_schedule.c (hunk 1 above)
Comment (lyakh) [reply src/schedule/zephyr_dp_schedule.c:134]: the same pointer should be used for freeing, that was allocated. If it's a cached pointer, so far our solution is to type-case it to uncached, e.g. like in https://github.com/thesofproject/sof/blob/1319163ea2c6d8bea6fd7189dea6275ac5e34bf5/src/audio/module_adapter/module_adapter.c#L403

File: src/schedule/zephyr_dp_schedule.c (hunk 2 above)
Comment (lyakh) [reply src/schedule/zephyr_dp_schedule.c:348]: Use this for `rballoc()` https://github.com/thesofproject/sof/blob/1319163ea2c6d8bea6fd7189dea6275ac5e34bf5/src/audio/module_adapter/module_adapter.c#L326-L327 and add `__sparse_cache` to `p_stack` definition

**Review by lyakh — CHANGES_REQUESTED** [review]

**Comment by marc-hb** [comment]
This makes https://github.com/thesofproject/sof/actions/runs/4433211392/jobs/7778032254?pr=7286  pass.

**Review by kv2019i — APPROVED** [review]

**Review by marcinszkudlinski — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**